src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
I resolved rate limit issues long time ago, WAIT is not only one available rate limit handler. I don't understand why are you spending so much time around making connections (the third PR to this plugin if i'm not mistaken).